### PR TITLE
Restore compatibility with < 4.07

### PR DESCRIPTION
**Pull request by smorimoto**
- The `Float` module requires  `ocaml >= 4.07`, so just replace with `string_of_float`
- dune requires `ocaml >= 4.02`, namely the minimum OCaml version requirement for this library is `>= 4.02`
- Travis has not been working for a long time and has been delayed in discovering this problem, so replace it with the GitHub Actions which is currently recommended

### CODE REVIEW

**Comment by smorimoto** [comment]
GitHub Actions will not actually be enabled in this repository until after this PR is merged. So, to check the latest build results, see here. You can confirm that everything is just ok: https://github.com/smorimoto/ocaml-cbor/actions/runs/1351401516

File: cbor.opam [hunk 1]
```diff
@@ -1,20 +1,41 @@
...
+build: [
+  ["dune" "subst"] {dev}
+  [
+    "dune"
+    "build"
+    "-p"
+    name
+    "-j"
+    jobs
+    "--promote-install-files=false"
```
Comment (ygrek) [inline cbor.opam:34]: I didn't see this one before, any particular reason? or is it new best practice? I guess it is related to the subsequent `dune install`?

File: dune-project [hunk 2]
```diff
@@ -1 +1,28 @@
-(lang dune 1.1)
+(lang dune 2.9)
```
Comment (ygrek) [inline dune-project:1]: can we keep lowest required dune version

**Review by ygrek — CHANGES_REQUESTED** [review]
Thanks!

**Comment by ygrek** [comment]
Rebased in bb51892
Thanks

**Pull request closed without merging**